### DISCUSSION TITLE
How do I move two databases into one database??

### DISCUSSION THREAD

**Discussion by JonMurphy** [teslamate-org/teslamate#3290] [Q&A]
Moved from here:
https://github.com/adriankumpf/teslamate/issues/3119#issuecomment-1643007387

After experiencing a password issue in February I ended up with two database, my original one and now a new one that started in March. I had rebuilt everything in February (near the original post) and I failed to notice. Can you point me in the right direction of how to combine two databases into one database?

There should be no duplicates for the trip stuff… DB stops and on a Feb date and DB starts on a March date.

**Comment by JonMurphy** [comment]
There is a lot more DB than I ever imagined!  First I grabbed the names of all of the tables.  And then went through each of them.

```
pi@teslamate:~/teslamate $ docker exec -it teslamate_database_1  bash
root@ad50f9bda514:/# 
root@ad50f9bda514:/# psql -U teslamate
psql (14.7 (Debian 14.7-1.pgdg110+1))
Type "help" for help.

teslamate=# 
teslamate=# \du
                                   List of roles
 Role name |                         Attributes                         | Member of 
-----------+------------------------------------------------------------+-----------
 teslamate | Superuser, Create role, Create DB, Replication, Bypass RLS | {}

teslamate=# \dt
                List of relations
 Schema |        Name        | Type  |   Owner   
--------+--------------------+-------+-----------
 public | addresses          | table | teslamate
 public | car_settings       | table | teslamate
 public | cars               | table | teslamate
 public | charges            | table | teslamate
 public | charging_processes | table | teslamate
 public | drives             | table | teslamate
 public | geofences          | table | teslamate
 public | positions          | table | teslamate
 public | schema_migrations  | table | teslamate
 public | settings           | table | teslamate
 public | states             | table | teslamate
 public | tokens             | table | teslamate
 public | updates            | table | teslamate
(13 rows)
```

Then I went through and grabbed info from each table:

```
# get the Column Names, etc. of the table
\d [tableName]

# how many rows are in this table?
SELECT count(*) from [tableName];

# get a sample (only 5 rows) of the data in the table
select * from [tableName] LIMIT 5;
```

Now I have a some appreciation for the work done to create TeslaMate!

**Comment by DrMichael** [comment]
I think, I have posted it already somewhere...

Here is the entity relationship diagram for teslamate.
![teslamate - ER diagram](https://github.com/adriankumpf/teslamate/assets/8087682/eb48155e-4938-4ff0-a2a4-37a57841684e)
